src/Router/RouteInvokableFactory.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Mvc\Router;
+
+use Interop\Container\ContainerInterface;
+use Zend\ServiceManager\Exception\ServiceNotCreatedException;
+use Zend\ServiceManager\Factory\AbstractFactoryInterface;
+
+/**
+ * Specialized invokable/abstract factory for use with RoutePluginManager.
+ *
+ * Can be mapped directly to specific route plugin names, or used as an
+ * abstract factory to map FQCN services to invokables.
+ */
+class RouteInvokableFactory implements AbstractFactoryInterface
+{
+    /**
+     * Can we create a route instance with the given name?
+     *
+     * Only works for FQCN $routeName values, for classes that implement RouteInterface.
+     *
+     * @param ContainerInterface $container
+     * @param string $routeName
+     * @return bool
+     */
+    public function canCreateServiceWithName(ContainerInterface $container, $routeName)
+    {
+        if (! class_exists($routeName)) {
+            return false;
+        }
+
+        if (! is_subclass_of($routeName, RouteInterface::class)) {
```

Comment:
We cannot use instanceof as $routeName is the string class name, not an
object instance.
On Oct 9, 2015 5:33 AM, "Abdul Malik Ikhsan" <notifications@github.com>
wrote:

> In src/Router/RouteInvokableFactory.php
> <https://github.com/zendframework/zend-mvc/pull/36#discussion_r41617243>:
>
> > +    /**
> > +     * Can we create a route instance with the given name?
> > +     *
> > +     * Only works for FQCN $routeName values, for classes that implement RouteInterface.
> > +     *
> > +     * @param ContainerInterface $container
> > +     * @param string $routeName
> > +     * @return bool
> > +     */
> > +    public function canCreateServiceWithName(ContainerInterface $container, $routeName)
> > +    {
> > +        if (! class_exists($routeName)) {
> > +            return false;
> > +        }
> > +
> > +        if (! is_subclass_of($routeName, RouteInterface::class)) {
>
> how about using $routeName instanceof RouteInterface ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-mvc/pull/36/files#r41617243>.
>